File: source/client/android_cili/app/src/main/java/com/vincent/android/cili/entity/Owner.kt
package com.vincent.android.cili.entity

import com.squareup.moshi.JsonClass
// import com.vincent.android.cili.annotation.NoArgAllOpen

// @NoArgAllOpen
@JsonClass(generateAdapter = true)
data class Owner(
    val name: String = "",
    val face: String = "",
    val fans: Int = 0,
    val id: Int = 0,
    val isFocus: Boolean = false
)

File: source/client/flutter_cili/lib/model/owner.dart
class Owner {
  final String name;
  final String face;
  final int fans;
  final int id;
  bool isFocus;

  Owner({
    required this.name,
    required this.face,
    required this.fans,
    required this.id,
    this.isFocus = false,
  });

  factory Owner.fromJson(Map<String, dynamic> json) => Owner(
    name: json['name'] ?? '',
    face: json['face'] ?? '',
    fans: json['fans'] ?? 0,
    id: json['id'] ?? 0,
    isFocus: json['isFocus'] ?? false,
  );

  Map<String, dynamic> toJson() => {
    'name': name,
    'face': face,
    'fans': fans,
    'id': id,
    'isFocus': isFocus,
  };
}
